syntax = "proto3";
package tinyim.auth;

option go_package = "./pb";

service AuthService {
  rpc Register (RegisterReq) returns (RegisterResp);
  rpc Login (LoginReq) returns (LoginResp);
  rpc Logout (LogoutReq) returns (LogoutResp);
}

message RegisterReq {
  string username = 1;
  string password = 2;
  string nickname = 3;
}

message RegisterResp {
  int64 user_id = 1;
  bool success = 2;
  string error_message = 3;
}

message LoginReq {
  string username = 1;
  string password = 2;
  string device = 3; // e.g. "PC", "Web", "Android"
}

message LoginResp {
  int64 user_id = 1;
  string token = 2;
  string nickname = 3;
  bool success = 4;
  string error_message = 5;
}

message LogoutReq {
  int64 user_id = 1;
  string token = 2;
  string device = 3;
}

message LogoutResp {
  bool success = 1;
}
